chainio/clients/avsregistry/builder.go

```diff
@@ -7,6 +7,61 @@ import (
 	"github.com/Layr-Labs/eigensdk-go/logging"
 )
 
+// Build an AVS registry client with the given configuration,
+// HTTP and WS clients, and logger, but without a private key.
+//
+// This is useful for read-only operations, such as fetching metrics.
+func BuildClientsForEcMetrics(
```

Comment:
This could be used for many other things than the economic collector metrics. Rename to `BuildReadClients`

Reply:
Done. I renamed it to `BuildReadClients`